code/modules/overmap/contacts/contact_sensors.dm

```diff
@@ -152,6 +144,28 @@
 			if(!record.pinged)
 				addtimer(CALLBACK(record, PROC_REF(ping)), time_delay)
 
+/**
+ * Adds an overmap object to the known contacts.
+ */
+/obj/machinery/computer/ship/sensors/proc/add_contact(obj/effect/overmap/contact)
+	var/datum/overmap_contact/record = new(src, contact)
+	var/bearing = get_bearing(contact)
+	contact_datums[contact] = record
+	if(contact.scannable)
+		playsound(loc, 'sound/machines/sensors/newcontact.ogg', 30, 1)
+		visible_message(SPAN_NOTICE("<b>\The [src]</b> states, \"New contact identified, designation [record.name], bearing [bearing].\""))
+	record.show()
+	animate(record.marker, alpha=255, 2 SECOND, 1, LINEAR_EASING)
+	return record
+
+/**
+ * Gets the bearing of an obj/effect/overmap/contact from src.
+ */
+/obj/machinery/computer/ship/sensors/proc/get_bearing(obj/effect/overmap/contact)
+	. = round(90 - Atan2(contact.x - linked.x, contact.y - linked.y),5)
+	if(. < 0)
+		. += 360
```

Comment:
We have a proc used by projectiles to get the bearings between two points, pixel-based, maybe use that?